userguide/aws/how-tos/add-a-certificate.adoc

```diff
@@ -0,0 +1,157 @@
+
+image::https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/aws/ELB-cert-validation-1.png[align=center]
+
+At the bottom you will see your domain name with a status of *Pending Validation*. Click the triangle next to your domain name. The display will expand to show the values that you must use to create your validation CNAME.
+
+image::https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/aws/ELB-cert-validation-2.png[align=center]
+
+The two strings labeled `Name` and `Value` should be copied to the two required entries for your CNAME. Copy these long strings carefully. Do not include any trailing periods. There is an option to save these strings in a file so they may be copied more easily, or stored for a later time.
+
+Your second CNAME is much simpler. The `Name` entry will be `kaui`. The `Value` entry will be simply your domain name, e.g., `mydomain.com`.
+
+=== 3. Create and Validate your Certificate
+
+When your CNAMEs are set, return to ACM and click *Continue*. The display changes to show that your certificate is *Pending Validation*. The certificate should now appear also on the main ACM page. When (and if) it is validated, the status will change to *Success*. *This may take several hours or more.* A short time later it will change again to *Issued*.
+
+When your certificate is successfully created, you may delete the temporary (validation) CNAME from your DNS Profile.
+
+
+== Using Let's Encrypt (with Certbot)
+
+`Let's Encrypt` is a root CA that is free to use. Certificates based on `Let's Encrypt` can be created using the application `certbot`. This approach is useful when using a third-party load balancer like `nginx`, which cannot be protected with ACM certificates. This is the load balancer used with the single-tier Kill Bill option.`Let's Encrypt` can be used free of charge.
+
+`Certbot` is pre-installed in the latest Kill Bill AMI, but if you need a different version of `certbot`, this is also available through the new package manager `snap`. This package manager is also preinstalled in the Kill Bill AMI.
+
+To setup a certificate using `certbot`, perform the following steps. This discussion assumes you are using `nginx`:
+
+=== 1. Create a CNAME
+
+Create a CNAME, following instructions from your domain provider. You will be asked to provide two items of information. The first is simply `kaui`. The second is the *Public IPV4 DNS* name for your EC2 instance, for example, `ec2-3-238-230-120.compute-1.amazonaws.com`. Note that the IPV4 address alone will *not* work.
+
+=== 2. Add your CNAME to the `nginx` configuration file
+
+This step tells `nginx` to forward HTTPS traffic via your CNAME.
+        
+First, login to your EC2 instance.
+
+Next, using a text editor of your choosing, edit the file `/etc/nginx/sites-enabled/killbill.conf`. You will need to use `sudo` to edit this file.
+
+This file contains two server blocks. The second block contains the lines:
+
+```
```

Comment:
Is there a problem here that should be fixed?

Reply:
@jdmooney - If you click on the "Show more details," you'll get information about the syntax problem. Instead of using ``` for code, the Asciidoctor syntax uses `[source]`. There's more information [here](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/).